src/output/file.rs

```diff
@@ -0,0 +1,289 @@
+// (c) Meta Platforms, Inc. and affiliates.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+use std::collections::BTreeMap;
+
+use crate::output as tv;
+use crate::spec;
+use maplit::{btreemap, convert_args};
+
+/// This structure represents a File message.
+/// ref: https://github.com/opencomputeproject/ocp-diag-core/tree/main/json_spec#file
+///
+/// # Examples
+///
+/// ## Create a File object with the `new` method
+///
+/// ```
+/// # use ocptv::output::*;
+///
+/// let file = File::new("name", "/path/to/file", false);
+/// ```
+///
+/// ## Create a File object with the `builder` method
+///
+/// ```
+/// # use ocptv::output::*;
+///
+/// let file = File::builder("name", "/path/to/file", false)
+///     .description("description")
+///     .content_type("text/plain")
+///     .add_metadata("key", "value".into())
+///     .build();
+/// ```
+#[derive(Default)]
+pub struct File {
+    name: String,
+    uri: String,
```

Comment:
should use a specific type instead of arbitrary strings; the spec says this is an URI, but only encodes it as a str because of the limitations of json schema. The user facing api here should not use strings.

example of what could be used (but we would need to re-export that as an Uri type instead of the name Url, because they're technically different -- that said, this url crate accepts URIs): https://docs.rs/url/latest/url/struct.Url.html#